x-pack/plugins/security_solution/server/endpoint/services/actions/action_files.ts

```diff
@@ -133,3 +135,84 @@ const doesFileHaveChunks = async (
 
   return Boolean((chunks.hits?.total as SearchTotalHits)?.value);
 };
+
+/**
+ * Returns subset of fileIds that don't have any file chunks
+ *
+ * @param esClient
+ * @param fileIds
+ * @returns fileIds
+ */
+export async function fileIdsWithoutChunks(
+  esClient: ElasticsearchClient,
+  fileIds: string[]
+): Promise<string[]> {
+  const noChunkFileIds = new Set(fileIds);
+
+  const chunks = await esClient.search<{ bid: string }>({
+    index: FILE_STORAGE_DATA_INDEX,
+    body: {
+      query: {
+        terms: {
+          'bid.keyword': fileIds,
+        },
+      },
+      _source: ['bid'],
+    },
+  });
+
+  chunks.hits.hits.forEach((hit) => {
+    const fileId = hit._source?.bid;
+    if (!fileId) return;
+    noChunkFileIds.delete(fileId);
+  });
+
+  return Array.from(noChunkFileIds);
+}
+
+/**
+ * Gets files with given status
+ *
+ * @param esClient
+ * @param status
+ */
+export function getFilesByStatus(esClient: ElasticsearchClient, status: FileStatus = 'READY') {
```

Comment:
Can you add a return type here.

Also - instead of returning the entire search results, maybe return only an array of files. I initially had a TS type defined for the file meta, but removed it because I did not think we needed it. You can bring it back (or a version of it) and just add it to the server code (does not need to go under `security_solution/common/endpoint` since its only used by the internal server processing).

The type I deleted is here: https://github.com/elastic/kibana/blob/5d2b2e59226ec18f5d4a840d1891c654a20d9850/x-pack/plugins/security_solution/common/endpoint/types/file_storage.ts